example/oven_fsm.dart

```diff
@@ -0,0 +1,226 @@
+/// Copyright (C) 2023 Intel Corporation
+/// SPDX-License-Identifier: BSD-3-Clause
+///
+/// oven_fsm.dart
+/// A simple oven FSM implementation using ROHD.
+///
+/// 2023 February 13
+/// Author: Yao Jing Quek <yao.jing.quek@intel.com>
+///
+
+// ignore_for_file: avoid_print
+
+// Import the ROHD package
+import 'package:rohd/rohd.dart';
+
+// Import the counter module implement in example.dart
+import './example.dart';
+
+// Enumerated type named `OvenStates` with four possible states:
+// `standby`, `cooking`,`paused`, and `completed`.
+enum OvenStates { standby, cooking, paused, completed }
+
+// One-hot encoded `Button` that extends from Const() value.
+// Represent start, pause, and resume as constant value `00`, `01`,
+// and `10` respectively.
+class Button extends Const {
+  Button._(int super.value) : super(width: 2);
+  Button.start() : this._(bin('00'));
+  Button.pause() : this._(bin('01'));
+  Button.resume() : this._(bin('10'));
+}
+
+// One-hot encoded `LEDLight` that extends from Const() value.
+// Represent yellow, blue, red and green LED as constant value `00`, `01`,
+// `10`, and `11` respectively.
+class LEDLight extends Const {
+  LEDLight._(int super.value) : super(width: 2);
+  LEDLight.yellow() : this._(bin('00'));
+  LEDLight.blue() : this._(bin('01'));
+  LEDLight.red() : this._(bin('10'));
+  LEDLight.green() : this._(bin('11'));
+}
+
+// Define a class OvenModule that extends ROHD's abstract Module class
+class OvenModule extends Module {
+  // A public variable with type StateMachine<OvenStates> `oven`.
+  // We want to return this variable to the main module for flexbility.
+  // Use `late` to indicate that the value will not be null
+  // and will be assign in the later section.
+  late StateMachine<OvenStates> oven;
+
+  // This oven module receives a `button` and a `reset` input from runtime.
+  OvenModule(Logic button, Logic reset) : super(name: 'OvenModule') {
+    // Register inputs and outputs of the module in the constructor.
+    // Module logic must consume registered inputs and output to registered
+    // outputs. `led` output also added as the output port.
+    button = addInput('button', button, width: button.width);
+    reset = addInput('reset', reset);
+    final led = addOutput('led', width: button.width);
+
+    // An internal clock generator
+    final clk = SimpleClockGenerator(10).clk;
+
+    // Register inputs and outputs, `counterReset` and `en` for internal signals
```

Comment:
If this is the registration of internal signals, then, I think, the text `Register inputs and outputs` is not needed? As far as I understand, registration is `addInput`/`addOutput`, but here it looks like `Define local signals` from <https://github.com/intel/rohd/blob/58584d2bc6ac0d7844dd92b2ecc85a5365bcfa5c/example/example.dart#L60-L62> I can be wrong.

Reply:
Now you mentioned it, it seems like a local signal to me. Let me change the wording.